src/openssl_lib.cpp

```diff
@@ -174,11 +184,15 @@ X509_REQ* OpenSSLLib::SSL_PEM_read_bio_X509_REQ(BIO *bp,
     return PEM_read_bio_X509_REQ(bp, x, cb, u);
 }
 
-BIO_METHOD* OpenSSLLib::SSL_BIO_s_mem() noexcept { return BIO_s_mem(); }
+const BIO_METHOD* OpenSSLLib::SSL_BIO_s_mem() noexcept { return BIO_s_mem(); }
 
 void OpenSSLLib::SSL_BIO_free_all(BIO* ptr) noexcept { BIO_free_all(ptr); }
 
-BIO* OpenSSLLib::SSL_BIO_new(BIO_METHOD* method) noexcept { return BIO_new(method); }
+BIO* OpenSSLLib::SSL_BIO_new(const BIO_METHOD* method) noexcept 
+{ 
+    //Const cast is needed in order to maintain retrocompatibility with OpenSSL 1.0.2 and older.
```

Comment:
Please create a new branch and provide a clean soltion. I want to get the code base cleaned up and not the maximum compatibility. OpenSSL 1.0.2x support will be dropped then.